src/PageHeader/PageHeader.tsx

```diff
@@ -0,0 +1,376 @@
+// PageHeader.ContextArea Sub Components: PageHeader.ParentLink, PageHeader.ContextBar, PageHeader.ContextAreaActions
+// ---------------------------------------------------------------------
+
+const ContextArea: React.FC<React.PropsWithChildren<PageHeaderProps>> = ({
+  children,
+  visible = onlyVisibleOnNarrowView,
+  sx = {}
+}) => {
+  const isVisible = useResponsiveValue(visible, false)
+  const contentNavStyles = {
+    display: isVisible ? 'flex' : 'none',
+    flexDirection: 'row',
+    alignItems: 'center',
+    gap: '8px',
+    order: REGION_ORDER.ContextArea
+  }
+  return <Box sx={merge<BetterSystemStyleObject>(contentNavStyles, sx)}>{children}</Box>
+}
+// adopted from React.AnchorHTMLAttributes
+type LinkProps = {
+  download?: string
+  href?: string
+  hrefLang?: string
+  media?: string
+  ping?: string
+  rel?: string
+  target?: string
+  type?: string
+  referrerPolicy?: React.AnchorHTMLAttributes<HTMLAnchorElement>['referrerPolicy']
+}
+export type ParentLinkProps = PageHeaderProps & LinkProps
+
+const ParentLink = React.forwardRef<HTMLAnchorElement, ParentLinkProps>(
+  (
+    {
+      children,
+      sx = {},
+      href,
+      'aria-label': ariaLabel = `Back to ${children}`,
+      as = 'a',
+      visible = onlyVisibleOnNarrowView
+    },
+    ref
+  ) => {
+    const isVisible = useResponsiveValue(visible, false)
+    return (
+      <>
+        <Link
+          ref={ref}
+          as={as}
+          aria-label={ariaLabel}
+          muted
+          sx={merge<BetterSystemStyleObject>(
+            {
+              display: isVisible ? 'flex' : 'none',
+              alignItems: 'center',
+              // min touch target size 44px (20 + (12x2))
+              lineHeight: '20px',
+              paddingY: '12px',
+              gap: '8px'
+            },
+            sx
+          )}
+          href={href}
+        >
+          <ArrowLeftIcon />
+          <Box>{children}</Box>
+        </Link>
+      </>
+    )
+  }
+) as PolymorphicForwardRefComponent<'a', ParentLinkProps>
+
+// ContextBar
+// Generic slot for any component above the title region. Use it for custom breadcrumbs and other navigation elements instead of ParentLink.
+// ---------------------------------------------------------------------
+
+const ContextBar: React.FC<React.PropsWithChildren<PageHeaderProps>> = ({
+  children,
+  sx = {},
+  visible = onlyVisibleOnNarrowView
+}) => {
+  const isVisible = useResponsiveValue(visible, false)
+  return <Box sx={merge<BetterSystemStyleObject>({display: isVisible ? 'flex' : 'none'}, sx)}>{children}</Box>
+}
+
+// ContextAreaActions
+// ---------------------------------------------------------------------
+const ContextAreaActions: React.FC<React.PropsWithChildren<PageHeaderProps>> = ({
+  children,
+  sx = {},
+  visible = onlyVisibleOnNarrowView
+}) => {
+  const isVisible = useResponsiveValue(visible, false)
```

Comment:
For visibility in layout, would it make sense to shift this kind of logic into CSS with breakpoints so that we don't have to do any extra work in SSR to predict the size and avoid a flash of change when JS is initialized?

Reply:
That makes so much sense! The way I implemented them is very much inspired by PageLayout and considering PageHeader will be used within PageLayout (I think most of the time) so it is important that we stay consistent for the DX. Having all said that, I see @colebemis has a [TODO](https://github.com/primer/react/blob/main/src/hooks/useResponsiveValue.ts#L48) for the `useResponsiveValue` to bring SSR support

I am happy to work on that enhancement. 

Are we thinking something like that for the SSR support or something different?

```
 // for narrow viewport
  '@media (max-width: 768px)': {
    display: hidden.narrow ? 'none' : 'flex'
  },
  // for regular viewport
  '@media (min-width: 768px) and (max-width: 1440px)': {
    display: hidden.regular ? 'none' : 'flex'
  },
  // for wide viewport
  '@media (min-width: 1440px)': {
    display: hidden.wide ? 'none' : 'flex'
  },

```

We could definitely optimise and make it reusable but I just wanted to gauge the feeling before going too further?

Reply:
Media queries vs the `useResponsiveValue` hook is a tradeoff between performance and developer ergonomics. In cases where we can use media queries easily, we should do that. In other cases, we can fall back to the `useResponsiveValue` hook. Does that make sense?

Re: SSR support for `useResponsiveValue` - I'm unsure how to make SRR work and keep the same API. Let me know if you have any ideas!

Reply:
```
 // for narrow viewport
  '@media (max-width: 768px)': {
    display: hidden.narrow ? 'none' : 'flex'
  },
  // for regular viewport
  '@media (min-width: 768px) and (max-width: 1440px)': {
    display: hidden.regular ? 'none' : 'flex'
  },
  // for wide viewport
  '@media (min-width: 1440px)': {
    display: hidden.wide ? 'none' : 'flex'
  },

```

A couple notes about this code snippet:

- The `regular` range includes the `wide` range too. So the `regular` media query would be `@media (min-width: 768px)`. [Reference](https://github.com/github/primer/blob/main/adrs/2022-04-15-responsive-design-api-guidelines.md#2-update-nomenclature-for-creating-responsive-components-and-web-pages)
- This would cause unexpected results if the user only specified `narrow` and `regular` keys (not `wide`): `hidden={{regular: true, narrow: false}}`. In that code snippet, `display` would be set to `flex` in the wide viewport. But the intended outcome should be that the component is still hidden on wide viewports. Does that make sense?

Reply:
I'll tackle this is a separate PR. I created an [issue](https://github.com/github/primer/issues/1569) for this 🙌🏼